server/controllers/RequestController.js

```diff
@@ -0,0 +1,351 @@
+            'Do you have any item that needs fixing? We love fixing stuff!',
+            result: [],
+          });
+        }
+        return res.status(200).json({
+          statusCode: 200,
+          error: [],
+          message: 'Your requests were succesfully retrieved',
+          result: result.rows,
+        });
+      })
+      .catch(() => res.status(500).json({
+        statusCode: 500,
+        error: 'Internal Server Error',
+        message: 'Oops! Another bug must have crawled into our systems, ' +
+          'but we are right on it! Please try your request later :\'(',
+      }));
+  }
+
+  static createRequest(req, res) {
+    // set auto-generated fields and create request object
+    const owner = req.user.email;
+    const request = new Request(
+      req.body.type, req.body.item, req.body.model,
+      req.body.detail, owner,
+    );
+
+    // save valid request to db
+    db.query(
+      `INSERT INTO requests (type, item, model, detail, status, owner, created_at) 
+      VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING *`,
+      [
+        request.type, request.item, request.model,
+        request.detail, request.status, request.owner, 'NOW()',
+      ],
+    )
+      .then((result) => {
+        if (result.rowCount < 1) {
+          return res.status(500).json({
+            statusCode: 500,
+            error: 'Internal Server Error',
+            message: 'Oops! Another bug must have crawled into our systems, ' +
+            'but we are right on it! Please try your request later :\'(',
+          });
+        }
+
+        return res.status(201).json({
+          statusCode: 201,
+          error: [],
+          message: 'Yay! Your request was successfuly created and is pending admin approval.',
+          request: {
+            id: result.rows[0].id,
+            type: result.rows[0].type,
+            item: result.rows[0].item,
+            model: result.rows[0].model,
+            detail: result.rows[0].detail,
+            createdAt: result.rows[0].created_at,
+          },
+        });
+      })
+      .catch((error) => {
+        res.status(500).json({
+          statusCode: 500,
+          error: 'Internal Server Error',
+          message: 'Oops! Another bug must have crawled into our systems, ' +
+          'but we are right on it! Please try your request later :\'(',
+        });
+        console.error('Error ', error.stack);
+      });
+  }
+
+  static getRequest(req, res) {
+    // get the request id
+    const { requestId } = req.params;
+
+    // query db for user request
+    db.query(
+      'SELECT * FROM requests WHERE owner = $1 AND id = $2',
+      [req.user.email, requestId],
+    )
+      .then((result) => {
+        if (result.rows < 1) {
+          return res.status(200).json({
+            statusCode: 500,
+            error: [],
+            message: 'You have no request with that id, please try another request id',
+            request: result.rows,
```

Comment:
Since you are assuming that the request has failed and there's no request with that id, why returning `request.rows`?